docs/asciidoc/modules/ROOT/pages/database-integration/vectordb/weaviate.adoc

```diff
@@ -0,0 +1,223 @@
+[source,cypher]
+----
+CALL apoc.vectordb.weaviate.upsert($host, 'test_collection',
+    [
+        {id: "8ef2b3a7-1e56-4ddd-b8c3-2ca8901ce308", vector: [0.05, 0.61, 0.76, 0.74], metadata: {city: "Berlin", foo: "one"}},
+        {id: "9ef2b3a7-1e56-4ddd-b8c3-2ca8901ce308", vector: [0.19, 0.81, 0.75, 0.11], metadata: {city: "London", foo: "two"}}
+    ],
+    {<optional config>})
+----
+
+
+.Get vectors (it leverages https://weaviate.io/developers/weaviate/api/rest#tag/objects/get/objects/\{className\}/\{id\}[this API])
+[source,cypher]
+----
+CALL apoc.vectordb.weaviate.get($host, 'test_collection', [1,2], {<optional config>})
+----
+
+
+.Example results
+[opts="header"]
+|===
+| score | metadata | id | vector | text | entity
+| null | {city: "Berlin", foo: "one"} | null | null | null | null
+| null | {city: "Berlin", foo: "two"} | null | null | null | null
+| ...
+|===
+
+
+.Get vectors with `{allResults: true}`
+[source,cypher]
+----
+CALL apoc.vectordb.weaviate.get($host, 'test_collection', [1,2], {allResults: true, <optional config>})
+----
+
+
+.Example results
+[opts="header"]
+|===
+| score | metadata | id | vector | text | entity
+| null | {city: "Berlin", foo: "one"} | 1 | [...] | null | null
+| null | {city: "Berlin", foo: "two"} | 2 | [...] | null | null
+| ...
+|===
+
+
+.Query vectors (it leverages https://weaviate.io/developers/weaviate/api/rest#tag/graphql/post/graphql[here])
+[source,cypher]
+----
+CALL apoc.vectordb.weaviate.query($host, 
+    'test_collection', 
+    [0.2, 0.1, 0.9, 0.7], 
+    '{operator: Equal, valueString: "London", path: ["city"]}', 
+    5, 
+    {fields: ["city", "foo"], allResults: true, <other optional config>})
+----
+
+
+.Example results
+[opts="header"]
+|===
+| score | metadata | id | vector | text
+| 1, | {city: "Berlin", foo: "one"} | 1 | [...] | null
+| 0.1 | {city: "Berlin", foo: "two"} | 2 | [...] | null
+| ...
+|===
+
+
+We can define a mapping, to fetch the associated nodes and relationships and optionally create them, by leveraging the vector metadata.
+
+For example, if we have created 2 vectors with the above upsert procedures,
+we can populate some existing nodes (i.e. `(:Test {myId: 'one'})` and `(:Test {myId: 'two'})`):
+
+
+[source,cypher]
+----
+CALL apoc.vectordb.weaviate.query($host, 'test_collection',
+    [0.2, 0.1, 0.9, 0.7],
+    {},
+    5, 
+    { fields: ["city", "foo"],
+      mapping: {
+        embeddingKey: "vect", 
+        nodeLabel: "Test", 
+        entityKey: "myId", 
+        metadataKey: "foo" 
+      }
+    })
+----
+
+which populates the two nodes as: `(:Test {myId: 'one', city: 'Berlin', vect: [vector1]})`
+and `(:Test {myId: 'two', city: 'London', vect: [vector2]})`,
+which will be returned in the `entity` column result.
+
+
+Or else, we can create a node if not exists, via `create: true`:
+
+[source,cypher]
+----
+CALL apoc.vectordb.weaviate.query($host, 'test_collection',
```

Comment:
Let's separate that into an extra procedure, e.g. updateGraph, so that the query method can remain read-only.